core/src/main/java/org/elasticsearch/index/mapper/ScaledFloatFieldMapper.java

```diff
@@ -504,7 +504,10 @@ public Index index() {
 
         @Override
         public NumericType getNumericType() {
-            return scaledFieldData.getNumericType();
+            /**
+             * {@link ScaledFloatLeafFieldData#getDoubleValues()} transforms the raw long values in `scaled` floats.
+             */
+            return NumericType.FLOAT;
```

Comment:
I don't think that will change anything in practice, but I think the correct value would be `DOUBLE`?